<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>742</width>
    <height>613</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Neural Network Test</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="controlsTab">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QWidget" name="horizontalWidget" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QGroupBox" name="DataBox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Data Group</string>
             </property>
             <property name="title">
              <string>DataSource</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <layout class="QGridLayout" name="gridLayout_5">
                <item row="0" column="2">
                 <widget class="QPushButton" name="inputOpenButton">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>135</width>
                    <height>24</height>
                   </size>
                  </property>
                  <property name="toolTip">
                   <string>Open Input File</string>
                  </property>
                  <property name="text">
                   <string>Open</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="inputLabel">
                  <property name="toolTip">
                   <string>Input File</string>
                  </property>
                  <property name="text">
                   <string>Input File</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="1" column="1" colspan="2">
                 <widget class="QLabel" name="currentFileName">
                  <property name="text">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="currentFileNameLabel">
                  <property name="text">
                   <string>Current File</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="ControlBox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Control Group</string>
             </property>
             <property name="title">
              <string>Control</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="0" column="0">
               <layout class="QGridLayout" name="gridLayout_4">
                <item row="1" column="0">
                 <widget class="QLabel" name="betaLabel">
                  <property name="toolTip">
                   <string>Stretch coefficient</string>
                  </property>
                  <property name="text">
                   <string>Neurons</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0">
                 <widget class="QLabel" name="precisionlabel">
                  <property name="toolTip">
                   <string>Precision</string>
                  </property>
                  <property name="text">
                   <string>Precision</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <spacer name="horizontalSpacer_11">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="2" column="2">
                 <widget class="QSpinBox" name="maxEpoch">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>134</width>
                    <height>23</height>
                   </size>
                  </property>
                  <property name="toolTip">
                   <string>Maximim number of epochs</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                  </property>
                  <property name="maximum">
                   <number>1000000</number>
                  </property>
                  <property name="singleStep">
                   <number>1</number>
                  </property>
                  <property name="value">
                   <number>500</number>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="alphaLabel">
                  <property name="toolTip">
                   <string>Training coefficient</string>
                  </property>
                  <property name="text">
                   <string>Alpha</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="maxEpochlabel">
                  <property name="toolTip">
                   <string>Maximim number of epochs</string>
                  </property>
                  <property name="text">
                   <string>Max epoch</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <spacer name="horizontalSpacer_10">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="1" column="2">
                 <layout class="QHBoxLayout" name="horizontalLayout_8">
                  <item>
                   <widget class="QSpinBox" name="numberOfNeurons">
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="maximum">
                     <number>500</number>
                    </property>
                    <property name="value">
                     <number>20</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="3" column="1">
                 <spacer name="horizontalSpacer_12">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="3" column="2">
                 <layout class="QHBoxLayout" name="horizontalLayout_9">
                  <item>
                   <widget class="QDoubleSpinBox" name="precisionMantiss">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="maximum">
                     <double>10.000000000000000</double>
                    </property>
                    <property name="value">
                     <double>1.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_5">
                    <property name="text">
                     <string>e</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="precisionDegree">
                    <property name="minimum">
                     <number>-16</number>
                    </property>
                    <property name="maximum">
                     <number>16</number>
                    </property>
                    <property name="value">
                     <number>-4</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="0" column="2">
                 <layout class="QHBoxLayout" name="horizontalLayout_6">
                  <item>
                   <widget class="QDoubleSpinBox" name="alphaMantiss">
                    <property name="decimals">
                     <number>3</number>
                    </property>
                    <property name="minimum">
                     <double>-1000.000000000000000</double>
                    </property>
                    <property name="maximum">
                     <double>1000.000000000000000</double>
                    </property>
                    <property name="singleStep">
                     <double>0.010000000000000</double>
                    </property>
                    <property name="value">
                     <double>1.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_3">
                    <property name="text">
                     <string>e</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="alphaDegree">
                    <property name="minimum">
                     <number>-16</number>
                    </property>
                    <property name="maximum">
                     <number>16</number>
                    </property>
                    <property name="value">
                     <number>-1</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="0" column="1">
                 <spacer name="horizontalSpacer_9">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="4" column="0">
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Test percent</string>
                  </property>
                 </widget>
                </item>
                <item row="4" column="1">
                 <spacer name="horizontalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item row="4" column="2">
                 <widget class="QDoubleSpinBox" name="doubleSpinBox">
                  <property name="decimals">
                   <number>2</number>
                  </property>
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="ResultsBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Results Group</string>
          </property>
          <property name="title">
           <string>Buttons</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QPushButton" name="startButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Start</string>
               </property>
               <property name="text">
                <string>Start</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="stopButton">
               <property name="toolTip">
                <string>Stop</string>
               </property>
               <property name="text">
                <string>Stop</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="saveButton">
               <property name="toolTip">
                <string>Save results to file</string>
               </property>
               <property name="text">
                <string>Save results</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="graphTab">
       <attribute name="title">
        <string>Graph</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QWidget" name="graphLayout" native="true">
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <widget class="QwtPlot" name="qwtPlot">
             <property name="toolTip">
              <string>Graph of results error</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <spacer name="horizontalSpacer_13">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="saveImageButton">
               <property name="text">
                <string>Save As Image</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>742</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAuthors"/>
   </widget>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionInput_File">
   <property name="text">
    <string>Input File</string>
   </property>
  </action>
  <action name="actionOutput_File">
   <property name="text">
    <string>Output File</string>
   </property>
  </action>
  <action name="actionAuthors">
   <property name="text">
    <string>Authors</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>inputOpenButton</tabstop>
  <tabstop>maxEpoch</tabstop>
  <tabstop>startButton</tabstop>
  <tabstop>stopButton</tabstop>
  <tabstop>saveButton</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>saveImageButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>stopButton</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>465</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>459</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>saveButton</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>469</y>
    </hint>
    <hint type="destinationlabel">
     <x>189</x>
     <y>459</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>stopButton</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>464</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>startButton</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>34</x>
     <y>466</y>
    </hint>
    <hint type="destinationlabel">
     <x>55</x>
     <y>468</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>startButton</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>127</x>
     <y>471</y>
    </hint>
    <hint type="destinationlabel">
     <x>67</x>
     <y>458</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>saveButton</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>461</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>463</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
